lib/cfa/grub2/grub_cfg.rb

```diff
@@ -12,8 +12,24 @@ class GrubCfg < BaseModel
       # @private only internal parser
       class Parser
         def self.parse(string)
-          menu_lines = string.lines.grep(/menuentry\s*'/)
-          menu_lines.map { |line| line[/\s*menuentry\s*'([^']+)'.*/, 1] }
+          result = []
+          submenu = ""
+          string.lines.each do |line|
+            case line
+            when /menuentry\s*'/
+              entry = line[/\s*menuentry\s+'([^']+)'.*/, 1]
+              result << {
+                title: entry,
+                path: submenu.empty? ? entry : "#{submenu}>#{entry}"
+              }
+            when /^}\s*\n/
+              submenu = ""
+            when /submenu\s/
+              submenu = line[/\s*submenu\s+'([^']+)'.*/, 1]
```

Comment:
currently it is not generated. I intentionally postpone solving such issue and hopefully meanwhile someone write augeas lense :)